### ISSUE TITLE
"IndexError: list index out of range" when viewing shopping list for meal plan containing days without a recipe selected

### ISSUE THREAD

**Issue by bjostad** [hay-kot/mealie#178]
**Describe the bug**
In our meal plan we left a day with no recipe selected (leftovers). This results in a blank shopping list if you attempt to view one for that meal plan.

**Steps To Reproduce**
1. Create Meal Plan that contains at least one day with no recipe selected.
2. View planner
3. Select shopping list in planner

**Expected behavior**
Shopping list generated

**Actual Behavior**
Blank shopping list dialog box. Header exists but there is no content.

**Device Information (please complete the following information):**
 - OS: Ubuntu 20.04
 - Deployment: Docker
 - Browser: Chrome mobile and desktop
 - Version: 0.3.0-dev

**Additional context**
This only occurs if you have a day in the meal plan that does not have a recipe selected. We resolved this by creating a recipe called "leftovers" as a filler for days we aren't planning to cook.

```INFO:     192.168.1.194:0 - "GET /api/meal-plans/all HTTP/1.1" 200 OK,
INFO:     192.168.1.194:0 - "POST /api/recipes/category HTTP/1.1" 200 OK,
INFO:     192.168.1.194:0 - "GET /api/recipes/dinner-at-v-e-s/image HTTP/1.1" 200 OK,
INFO:     192.168.1.194:0 - "GET /api/recipes/null/image HTTP/1.1" 200 OK,
INFO:     192.168.1.194:0 - "GET /api/meal-plans/1/shopping-list HTTP/1.1" 500 Internal Server Error,
ERROR:    Exception in ASGI application,
Traceback (most recent call last):,
  File "/usr/local/lib/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 398, in run_asgi,
    result = await app(self.scope, self.receive, self.send),
  File "/usr/local/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__,
    return await self.app(scope, receive, send),
  File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 199, in __call__,
    await super().__call__(scope, receive, send),
  File "/usr/local/lib/python3.9/site-packages/starlette/applications.py", line 111, in __call__,
    await self.middleware_stack(scope, receive, send),
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__,
    raise exc from None,
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__,
    await self.app(scope, receive, _send),
  File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__,
    raise exc from None,
  File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__,
    await self.app(scope, receive, sender),
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 566, in __call__,
    await route.handle(scope, receive, send),
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 227, in handle,
    await self.app(scope, receive, send),
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 41, in app,
    response = await func(request),
  File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 201, in app,
    raw_response = await run_endpoint_function(,
  File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 150, in run_endpoint_function,
    return await run_in_threadpool(dependant.call, **values),
  File "/usr/local/lib/python3.9/site-packages/starlette/concurrency.py", line 34, in run_in_threadpool,
    return await loop.run_in_executor(None, func, *args),
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run,
    result = self.fn(*self.args, **self.kwargs),
  File "./routes/meal_routes.py", line 26, in get_shopping_list,
    recipes = [db.recipes.get(session, x) for x in slugs],
  File "./routes/meal_routes.py", line 26, in <listcomp>,
    recipes = [db.recipes.get(session, x) for x in slugs],
  File "./db/db_base.py", line 111, in get,
    return db_entries[0],
IndexError: list index out of range,```

**Comment by hay-kot** [comment]
There was no check in the loop that builds the ingredients list to check if it wasn't NoneType. I added a check to determine if the recipe is in that slot which fixed the issue. Will be fixed in the next update 👍

**Comment by hay-kot** [comment]
Fixed in [v0.4.0](https://hay-kot.github.io/mealie/changelog/v0.4.0/)
